src/Runner.jl

```diff
@@ -351,6 +351,9 @@ function generate_compiler_wrappers!(platform::AbstractPlatform; bin_path::Abstr
                 min_macos_version_flag(p),
             ])
         end
+        if Sys.isfreebsd(p)
```

Comment:
Can you please add a comment about why this is needed?  I believe you still need to do the the same for GCC on musl-based systems

Reply:
Indeed! I've added a comment now, please proof read. Also, I had to replace `$(prefix)` by `/workspace/destdir` -- if you know a better way to avoid duplicating this here, please let me know.

I don't think we need this for GCC on musl-based systems? As far as I recall, only GCC 4.8.5 on musl had the issue, but I fixed that in https://github.com/JuliaPackaging/Yggdrasil/pull/3969

Anyway: it'll be trivial to add, so perhaps we can first test this version on Yggdrasil, and then add it if we determine it is needed after all?

Reply:
> but I fixed that in https://github.com/JuliaPackaging/Yggdrasil/pull/3969

I don't think we rebuilt GCC after that, did we?

Reply:
Oh, somehow I thought it was. My bad then. In any case, I just opened https://github.com/JuliaPackaging/Yggdrasil/pull/4523 as I think it's best to establish a baseline now, then use that to test this PR later on.

Reply:
Interestingly, that PR worked on FreeBSD... But it indeed failed with musl. I've now added a few more packages to that test, and also packages that do `LDFLAGS="-L${libdir}"` -- perhaps we need something similar for that, too?

Reply:
I keep adding to that test PR, removing `-I${includedir}` statements, and so far only musl builds were broken, no FreeBSD 😂 but I know that it recently was an issue for one of the GAP_pkg_*_jlls, so it definitely still is there (I'll try to repro it now with one of those).

There are also various mingw failures, but I think they are unrelated?

Reply:
Updated to check for musl